arch/riscv/Kconfig

```diff
@@ -290,6 +290,16 @@ config RISCV_GENERIC_TOOLCHAIN
 	  Allow SOCs that have custom extended riscv ISA to still
 	  compile with generic riscv32 toolchain.
 
+config USE_ISR_WRAPPER
+	bool "Use isr_wrapper to handle interrupt and/or exception/fault"
+	default y if GEN_SW_ISR_TABLE
+	default y if MULTITHREADING
+	help
+	  This is helper config to be able to use exception handling from
+	  _isr_wrapper when GEN_SW_ISR_TABLE is not enabled. E.g. MULTITHREADING
+	  needs exception handling and thread entry/switch functions but may
+	  use only irq vector table.
+
```

Comment:
Since the GEN_SW_ISR_TABLE depends on the wrapper, GEN_SW_ISR_TABLE should select USE_ISR_WRAPPER. This option as it is now can be disabled `CONFIG_USE_ISR_WRAPPER=n` regardless of GEN_SW_ISR_TABLE and MULTITHREADING, which would break :) no reason to give anyone the option

Reply:
Please check if it is right now ?